lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -59,13 +59,15 @@ def collection_ancestor? collections
     end
 
     def ancestor? collection
-      return true if collection == self
+      return true if (( !collection.id.nil? && !self.id.nil? ) && ( collection.id == self.id ))
+      return true if (( collection.id.nil? || self.id.nil? ) && ( collection == self ))
```

Comment:
Of course, thinking this through a little further.  Even the way the core.rb method is written, if the full object equivalence fails, it can still return true if the class and id match and it isn't a new record.  If it is ok for this method to return true even if the state of self and the comparison_object don't match, then it can be re-written with the faster check first.